sif/internal/sif_util.go

```diff
@@ -0,0 +1,254 @@
+package internal
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/opencontainers/go-digest"
+	"github.com/pkg/errors"
+	uuid "github.com/satori/go.uuid"
+	"github.com/yhcote/sif/pkg/sif"
```

Comment:
This is a fork of a repo that claims to be a “reference implementation” of SIF. If that’s true, it would be ideal (I have no idea if feasible) to actually use that implementation, possibly contributing improvements there, rather than commit c/image to maintaining a long-term internal fork (“This branch is 314 commits behind hpcng:master.”)

(I’m afraid I haven’t yet reviewed this PR, nor the changes in that fork, so I don’t have an opinion on what’s the right approach yet, at all. At this point I just want to make sure the trade-offs are recognized.)